src/MongoDB/Manager.c

```diff
@@ -805,6 +805,18 @@ static PHP_METHOD(Manager, startSession)
 		}
 	}
 
+	if (options && php_array_existsc(options, "snapshot")) {
+		if (!cs_opts) {
+			cs_opts = mongoc_session_opts_new();
+		}
+		mongoc_session_opts_set_snapshot(cs_opts, php_array_fetchc_bool(options, "snapshot"));
+	}
+
+	if (cs_opts && mongoc_session_opts_get_causal_consistency(cs_opts) && mongoc_session_opts_get_snapshot(cs_opts)) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "Only one of \"causalConsistency\" and \"snapshot\" can be enabled");
```

Comment:
libmongoc also raises this error through mongoc_client_start_session; however, the code/domain used for that would lead to a PHPC throwing a RuntimeException. I handled this manually so we can ensure an InvalidArgumentException is thrown, which makes more sense since this is just option validation.